flake.nix

```diff
@@ -0,0 +1,20 @@
+{
+  description = "CLI for Hosted Nix binary caches";
+
+  outputs = { self }: let
+    systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "aarch64-linux" ];
+    forAllSystems = f: builtins.listToAttrs (map (name: { inherit name; value = f name; }) systems);
+  in {
+    packages = forAllSystems (system: {
+      cachix = let
+        pkgs = import ./nix {
+          inherit system;
+        };
+        cachix-api = pkgs.haskellPackages.callCabal2nix "cachix-api" (pkgs.gitignoreSource ./cachix-api) {};
```

Comment:
Is there any way to deduplicate this against default.nix? I'll surely forget to update it the next time.

On top of that CI should also build the flake bit I guess, so maybe we could use flake compat?

Reply:
If I understand it right, with flakes there isn't much need in `niv` and it should be possible move all external dependencies into `inputs` attribute of `flake.nix`. The specific revisions would be pinned in `flake.lock`